### ISSUE TITLE
fail_level: error (or any level above none) makes the job fail on the suggestion step

### ISSUE THREAD

**Issue by anatox** [reviewdog/action-shellcheck#100]
### What happens

If you set `fail_level` to anything other than none (error, warning, info, any), the action exits non-zero on the Running shellcheck (suggestion) step even when the main "Running shellcheck" step succeeded. The main step is fine; the suggestion pass breaks the workflow.

### How to reproduce

```bash
# testdata/test.sh
#!/usr/bin/env bash
echo $1 # Unquoted variables
find . -name *.txt # Unquoted find/grep patterns
```

```yaml
- uses: reviewdog/action-shellcheck@v1
  with:
    fail_level: error # any value other than `none` triggers the bug
    level: warning
    filter_mode: nofilter
    reporter: github-pr-review
```

### Expected
Suggestion pass should always succeed (besides internal errors).

### Actual output

Action exits with code 1, and the job fails - even though there are no errors reported by shellcheck

```text
::group:: Running shellcheck ...
testdata/test.sh:2:6: info: Double quote to prevent globbing and word splitting. (ShellCheck.SC2086)
testdata/test.sh:3:14: warning: Quote the parameter to -name so the shell won't interpret it. (ShellCheck.SC2061)
testdata/test.sh:3:14: info: Use ./*glob* or -- *glob* so names with dashes won't become options. (ShellCheck.SC2035)
::endgroup::
::group:: Running shellcheck (suggestion) ...
testdata/test.sh:2:-echo $1 # Unquoted variables
testdata/test.sh:2:+echo "$1" # Unquoted variables
reviewdog: found at least one issue with severity greater than or equal to the given level: error
::endgroup::
```

**Comment by grische** [comment]
Thanks for tracking this down, the suggested `-fail-level=none` does fix the symptom for the suggestion pass.

I think that this has a different root cause though:

[`parser/parser.go:62-74`](https://github.com/reviewdog/reviewdog/blob/v0.21.2/parser/parser.go#L62-L74) maps tool severities:

```go
case "error", "ERROR", "Error", "e", "E":       return rdf.Severity_ERROR
case "warning", "WARNING", "Warning", "w", "W": return rdf.Severity_WARNING
case "info", ..., "note", ...:                  return rdf.Severity_INFO
default:                                        return rdf.Severity_UNKNOWN_SEVERITY
```

There is no case for shellcheck's `style` level (i.e. triggering UNKNOWN severity) and
with [`FailLevel.ShouldFail`](https://github.com/reviewdog/reviewdog/blob/v0.21.2/fail_level.go#L67-L73):

```go
return minSeverity == rdf.Severity_UNKNOWN_SEVERITY || severity <= minSeverity
```

This causes it to fire for every non-`none` fail level.
